sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/util/ComposeUtil.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.sample_compose_fragments.util
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.flowWithLifecycle
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.StateFlow
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+object ComposeUtil {
```

Comment:
Do we need these custom extension functions? Can we use the `.collectAsStateWithLifecycle()` function instead? https://developer.android.com/jetpack/compose/state#use-other-types-of-state-in-jetpack-compose

Reply:
Yeah we can use `.collectAsStateWithLifecycle()` that function, one minor benefit I can see:

The benefit of using `rememberFlow` and `collectAsStateLifecycleAware` instead of `collectAsStateWithLifecycle` is that it allows you to reuse the flow across multiple composables. This is because `rememberFlow` creates a new flow that is scoped to the lifecycle of the composable that it is used in. This means that the flow will only be active while the composable is visible, and it will be automatically disposed of when the composable is destroyed.

`collectAsStateWithLifecycle` on the other hand, creates a new flow for each composable that it is used in. This means that if you use `collectAsStateWithLifecycle` in multiple composables, you will end up with multiple flows that are all tracking the same data. This can lead to memory leaks and other problems.